Source/ARTRealtimeChannel.m

```diff
@@ -321,6 +321,10 @@ - (void)transition:(ARTRealtimeChannelState)state status:(ARTStatus *)status {
         [_attachedEventEmitter emit:[NSNull null] with:[ARTErrorInfo createWithCode:90000 message:msg]];
     }
 
+    [self unlessStateChangesBefore:self.realtime.options.channelRetryTimeout do:^{
```

Comment:
You're unconditionally re-attaching here after every channel transition? This has nothing to do with the logic specified at RTL13a and RTL13b.